### ISSUE TITLE
Relocated store URI is reported as `'local'`, wrong

### ISSUE THREAD

**Issue by roberth** [NixOS/nix#11873]
**Describe the bug**

If you have a problem with a specific package or NixOS,
you probably want to file an issue at https://github.com/NixOS/nixpkgs/issues. 

**Steps To Reproduce**

There's probably an easier way to trigger this.

1. Choose or make an unimportant store path `<path>`
1. Back up a store path with `nix copy --to ~/recoverstore <path>`
1. Corrupt the original store path by mounting the system store file system `rw` and deleting it from the file system. Do not delete it from the store db; we want that entry to remain.
1. `sudo nix store repair --extra-substituters ~/recoverstore --extra-trusted-substituters ~/recoverstore <path>` (the latter option seems to have no effect; need `?trusted=true`; different issue)

Observe a message like

```
warning: ignoring substitute for '/nix/store/nzb7spdhiy30ikv1crnk8dgl0svyjgw4-libunistring-1.2' from 'local', as it's not signed by any of the keys in 'trusted-public-keys'
```

**Expected behavior**

```
warning: ignoring substitute for '/nix/store/nzb7spdhiy30ikv1crnk8dgl0svyjgw4-libunistring-1.2' from '/root/recoverstore', as it's not signed by any of the keys in 'trusted-public-keys'
```

**`nix-env --version` output**

**Additional context**

**Priorities**

Add :+1: to [issues you find important](https://github.com/NixOS/nix/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc).

**Comment by Ericson2314** [comment]
Yeah I think it doesn't render query params, and that is the normal form. Maybe my store reference cleanup branches can help with this...
